superset/assets/src/explore/components/controls/DateFilterControl.jsx

```diff
@@ -28,172 +51,265 @@ const propTypes = {
-          >
-            <InputGroup bsSize="small">
-              <InputGroup.Addon>
-                <Glyphicon glyph="calendar" />
-              </InputGroup.Addon>
-              <Datetime
-                inputProps={{ className: 'form-control input-sm' }}
-                dateFormat="YYYY-MM-DD"
-                defaultValue={this.state.dttm}
-                onFocus={this.setType.bind(this, 'fix')}
-                onChange={this.setDatetime.bind(this)}
-                timeFormat="h:mm:ss"
-              />
-            </InputGroup>
-          </PopoverSection>
-          <PopoverSection
-            title="Relative"
-            isSelected={this.state.type === 'rel'}
-            onSelect={this.setType.bind(this, 'rel')}
-          >
-            <div className="clearfix">
-              <div style={{ width: '50px' }} className="input-inline">
-                <FormControl
-                  onFocus={this.setType.bind(this, 'rel')}
-                  value={this.state.num}
-                  onChange={this.onNumberChange.bind(this)}
-                  bsSize="small"
-                />
-              </div>
-              <div style={{ width: '95px' }} className="input-inline">
-                <Select
-                  onFocus={this.setType.bind(this, 'rel')}
-                  value={this.state.grain}
-                  clearable={false}
-                  options={TIME_GRAIN_OPTIONS.map(s => ({ label: s, value: s }))}
-                  onChange={this.onControlChange.bind(this, 'grain')}
-                />
-              </div>
-              <div style={{ width: '95px' }} className="input-inline">
-                <Select
-                  value={this.state.rel}
-                  onFocus={this.setType.bind(this, 'rel')}
-                  clearable={false}
-                  options={RELATIVE_TIME_OPTIONS.map(s => ({ label: s, value: s }))}
-                  onChange={this.onControlChange.bind(this, 'rel')}
-                />
-              </div>
-            </div>
-          </PopoverSection>
-          <PopoverSection
-            title="Free form"
-            isSelected={this.state.type === 'free'}
-            onSelect={this.setType.bind(this, 'free')}
-            info={
-              'Superset supports smart date parsing. Strings like `last sunday` or ' +
-              '`last october` can be used.'
-            }
+          <Tabs
+            defaultActiveKey={TYPES.indexOf(this.state.type) + 1}
+            id="type"
+            bsStyle="pills"
+            onSelect={type => this.setState({ type: TYPES[type - 1] })}
           >
-            <FormControl
-              onFocus={this.setType.bind(this, 'free')}
-              value={this.state.free}
-              onChange={this.onFreeChange.bind(this)}
-              bsSize="small"
-            />
-          </PopoverSection>
+            <Tab eventKey={1} title="Range">
+              <FormGroup>
+                {timeFrames}
```

Comment:
maybe this should be its own third tab? It seems like the Range tab is a little busy with the radio buttons and also the dropdown menus

Reply:
@vylc, any thoughts here? Personally I like this design, since the common ranges and the dropdown serve the same purpose, and allow for a lot of control in a single place.